analog_forecast.py

```diff
@@ -0,0 +1,297 @@
+        )
+        search_da = search_da.where(~search_da.time.isin(ref_window), drop=True)
+        
+    return search_da
+
+
+def rmse(da, ref_da):
+    """Compute the RMSE between each time slice in da and ref_date
+    
+    Args:
+        da (xarray.DataArray): data array with time, latitude, and longitude dims for computing RMSE over the spatial dimensions
+        ref_da (xarray.DataArray): reference data array with only spatial axes (2d grid) to compute RMSE against
+        
+    Returns:
+        rmse_da (xarrya.DataArray): data array where each time step is the RMSE between da at that same time step and ref_da
+    """
+    try:
+        rmse_arr = np.sqrt(np.apply_over_axes(np.nanmean, (da - ref_da) ** 2, axes=[1,2]).squeeze())
+    except MemoryError:
+        print("Encountered MemoryError. Retrying with dask")
+        sub_da = sub_da.chunk(time=1)
+        sq_err = (sub_da - ref_da) ** 2
+        rmse_arr = np.sqrt(np.nanmean(sq_err, axis=(1,2)))
+    
+    rmse_da = xr.DataArray(
+        data=rmse_arr,
+        dims=["time"],
+        coords=dict(
+            time=da.time.values,
+        ),
+    )
+    
+    return rmse_da
+
+
+def run_rmse_over_time(da, ref_date, window):
+    """Get the RMSE between da at ref_time and every other preceding time slice. Calls compute() to process with dask.
+    
+    Args:
+        da (xarray.DataArray): ERA5 dataArray, with time, latitude, longitude axes
+        ref_date (str): reference date to compare all other preceding time slices against
+        window (str): method for temporal search, passed to get_search_da
+        
+    Returns:
+        rmse_da (xarra.DataArray): data array with an RMSE computed for each date preceding ref_date
+    """
+    ref_da = da.sel(time=ref_date).squeeze()
+    search_da = get_search_da(da, ref_date, window)
+    rmse_da = rmse(search_da, ref_da)
+    
+    return rmse_da
+
+
+def take_analogs(error_da, buffer, ref_date, n=5):
+    """Take the top n times from error_da ('top' == smallest error) with constraints on how close in time these top times may be.
+    
+    Args:
+        error_da (xarray.DataArray): error values indexed by time
+        buffer (int): number of days to buffer each top time for excluding subsequent times
+        ref_date (str): 
+        n (int): number of analogs to keep
+        
+    Returns:
+        analogs (xarray.DataArray): top n analogs taken from error_da
+    """
+    # sort ascending
+    error_da = error_da.sortby(error_da)
+    # iterate over list of top analogs and create a window that others must not overlap, lest they be discarded
+    analog_buffer_ranges = []
+    analog_times = []
+    td_buffer = pd.to_timedelta(buffer, "d")
+    for t in error_da.time.values:
+        if not any([t in dr for dr in analog_buffer_ranges]):
+            analog_buffer_ranges.append(pd.date_range(t - td_buffer, t + td_buffer))
+            analog_times.append(t)
+        if len(analog_buffer_ranges) == 5:
```

Comment:
limit of 5 analogues is hard coded here...consider just using the "n" parameter from the ```take_analogs``` function?